ml/src/main/scala/org/apache/spark/ml/h2o/param/H2OTargetEncoderParams.scala

```diff
@@ -0,0 +1,125 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.spark.ml.h2o.param
+
+import org.apache.spark.h2o.utils.H2OSchemaUtils
+import org.apache.spark.ml.h2o.features._
+import org.apache.spark.ml.PipelineStage
+import org.apache.spark.ml.param.{Param, Params, StringArrayParam}
+import org.apache.spark.sql.types.{DoubleType, StructField, StructType}
+import org.apache.spark.h2o.Frame
+
+trait H2OTargetEncoderParams extends PipelineStage with Params {
+
+  //
+  // Override pipeline stage members
+  //
+  override def transformSchema(schema: StructType): StructType = {
+    val flatSchema = H2OSchemaUtils.flattenSchema(schema)
+    require(getLabelCol() != null, "Label column can't be null!")
+    require(getInputCols() != null && getInputCols().nonEmpty, "The list of input columns can't be null or empty!")
+    val fields = flatSchema.fields
+    val fieldNames = fields.map(_.name)
+    require(fieldNames.contains(getLabelCol()),
+      s"The specified label column '${getLabelCol()}' was not found in the input dataset!")
+    for(inputCol <- getInputCols()) {
+      require(fieldNames.contains(inputCol),
+        s"The specified input column '$inputCol' was not found in the input dataset!")
+    }
+    val ioIntersection = getInputCols().intersect(getOutputCols())
+    require(ioIntersection.isEmpty,
+      s"""The columns [${ioIntersection.map(i => s"'$i'").mkString(", ")}] are specified
+         |as input columns and also as output columns. There can't be an overlap.""".stripMargin)
+    StructType(flatSchema.fields ++ getOutputCols().map(StructField(_, DoubleType, nullable = true)))
+  }
+
+  //
+  // List of Parameters
+  //
+  protected final val foldCol = new NullableStringParam(this, "foldCol", "Fold column name")
+  protected final val labelCol = new Param[String](this, "labelCol", "Label column name")
+  protected final val inputCols = new StringArrayParam(this, "inputCols", "Names of columns that will be transformed")
+  protected final val holdoutStrategy = new H2OTargetEncoderHoldoutStrategyParam(this,
+    "holdoutStrategy",
+    """A strategy deciding what records will be excluded when calculating the target average on the training dataset.
+      |Options:
+      | None        - All rows are considered for the calculation
+      | LeaveOneOut - All rows except the row the colculation is made for
+      | KMeans      - Only out-of-fold data is considered (The option requires foldCol to be set.)
+    """.stripMargin)
+  protected final val blending = new CaseClassParam[H2OTargetEncoderBlendingSettings](this,
+    "blending",
+    """If set, the target average becomes a weighted average of the group target value and the global target value of a given row.
```

Comment:
@mn-mikke  maybe like this -> "....average of the posterior average for a given categorical level and the prior average of the target."
"global target value of a given row" - does not make sense as global value is not being computed for a row.